### ISSUE TITLE
install.sh (v2.13.1): checksum verification fails — script downloads the .sbom.json asset instead of the linux-amd64 tarball

### ISSUE THREAD

**Issue by srgvg** [golangci/golangci-lint#6751]
## Description

Running the official install script pinned to `v2.13.1` fails checksum verification on
`linux/amd64`:

```
golangci/golangci-lint info checking GitHub for tag 'v2.13.1'
golangci/golangci-lint info found version: 2.13.1 for v2.13.1/linux/amd64
golangci/golangci-lint err hash_sha256_verify checksum for '/tmp/tmp.XXXXXXXXXX/golangci-lint-2.13.1-linux-amd64.tar.gz' did not verify b17bfbc9d4aaa48be7f4f1ce3240bc3d8200c870c072bacf15c26219e2cfb9cc
fe0b55fc4464b71bcdb9fae1bc1b0cbc8ccf2d1b5ad372e178fdded14f83ce67 vs b17bfbc9d4aaa48be7f4f1ce3240bc3d8200c870c072bacf15c26219e2cfb9cc
```

Reproduced identically and independently on two separate repositories' CI (different jobs,
different self-hosted GitHub Actions runners), and on a manual rerun of the same job — same two
hashes both times, not a transient/network flake.

## Root cause (confirmed against the release's own checksums file)

`https://github.com/golangci/golangci-lint/releases/download/v2.13.1/golangci-lint-2.13.1-checksums.txt`
contains:

```
b17bfbc9d4aaa48be7f4f1ce3240bc3d8200c870c072bacf15c26219e2cfb9cc  golangci-lint-2.13.1-linux-amd64.tar.gz
fe0b55fc4464b71bcdb9fae1bc1b0cbc8ccf2d1b5ad372e178fdded14f83ce67  golangci-lint-2.13.1-linux-amd64.tar.gz.sbom.json
```

The hash the install script actually downloaded and verified against
(`fe0b55fc...`) is the checksum of the **`.sbom.json` SBOM attestation file**, not the
`.tar.gz` binary archive. The script (or the asset-resolution logic it calls into) appears to be
matching `golangci-lint-2.13.1-linux-amd64.tar.gz` as a *prefix* of
`golangci-lint-2.13.1-linux-amd64.tar.gz.sbom.json` and fetching the wrong asset, rather than
requiring an exact filename match.

## Reproduction

```bash
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
  | sh -s -- -b /tmp/golangci-bin v2.13.1
```

on `linux/amd64`.

## Impact

Any CI or local environment pinning `golangci-lint = v2.13.1` via this install script and
performing a fresh install (no pre-existing cached binary) fails outright — this is a hard
install-time failure, not a lint-behavior issue.

## Expected

The script downloads and verifies `golangci-lint-2.13.1-linux-amd64.tar.gz` against
`b17bfbc9d4aaa48be7f4f1ce3240bc3d8200c870c072bacf15c26219e2cfb9cc`, its own correct checksum.

**Comment by ldez** [comment]
Duplicate of #6537, #6539, #6540, #6555, #6556, #6568, #6570, #6572, #6728, #6745.

---

> [!IMPORTANT] 
> If you are using the install script from the `master` branch:
> `https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh`
> This branch is not used anymore, we are using `main`.
>
> We recommend using `https://golangci-lint.run/install.sh` as URL to install golangci-lint.
> 
> https://golangci-lint.run/docs/welcome/install/local/#binaries

---
